src/directives/formly-field.js

```diff
@@ -152,13 +152,26 @@ function formlyField($http, $q, $compile, $templateCache, formlyConfig, formlyVa
     function resetModel() {
       $scope.model[$scope.options.key] = $scope.options.initialValue;
       if ($scope.options.formControl) {
-        $scope.options.formControl.$setViewValue($scope.model[$scope.options.key]);
-        $scope.options.formControl.$render();
-        $scope.options.formControl.$setUntouched();
-        $scope.options.formControl.$setPristine();
+        if(angular.isArray($scope.options.formControl)){
+          angular.forEach($scope.options.formControl, function(formControl){
+            resetFormControl(formControl, true);
+          });
+        } else {
+          resetFormControl($scope.options.formControl);
+        }
       }
     }
 
+    function resetFormControl(formControl, isMultiNgModel){
+      if(!isMultiNgModel){
+        formControl.$setViewValue($scope.model[$scope.options.key]);
+      }
```

Comment:
Actually, I think I understand. It's because we're explicitly setting the `$viewValue` without running it through `$formatters`? I would expect that angular does that for us when we call `$setViewValue` though... But I don't know.